pkg/aws/services/tagging.go

```diff
@@ -0,0 +1,97 @@
+package services
+
+import (
+	"context"
+	"github.com/aws/aws-application-networking-k8s/pkg/utils"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	taggingapi "github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	taggingapiiface "github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi/resourcegroupstaggingapiiface"
+)
+
+//go:generate mockgen -destination tagging_mocks.go -package services github.com/aws/aws-application-networking-k8s/pkg/aws/services Tagging
+
+const (
+	resourceTypePrefix = "vpc-lattice:"
+
+	ResourceTypeTargetGroup = resourceTypePrefix + "targetgroup"
+
+	maxArnsPerGetResourcesApi = 100
+)
+
+type Tags = map[string]*string
+
+type Tagging interface {
+	taggingapiiface.ResourceGroupsTaggingAPIAPI
+
+	// Receives a list of arns and returns arn-to-tags map.
+	GetTagsFromArns(ctx context.Context, arns []*string) (map[string]Tags, error)
```

Comment:
From -> For?

Reply:
arns []string, I dont think we need reference type here

Reply:
Also you can consider a strong typed version arn.ARN, much easier to use once constructed.

https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.47.9/aws/arn#Parse

```go
tgArns := utils.SliceMap(resp, func(tg *vpclattice.TargetGroupSummary) arn.ARN { 
  tgArn , _ := arn.Parse(*tg.Arn)
  return tgArn
})
```

Reply:
I would prefer string more, ARN is purely an identifier here and processing it adds no value.